internal/credential/vault/repository_ssh_certificate_credential_library_test.go

```diff
@@ -0,0 +1,1294 @@
+package vault
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/credential"
+	"github.com/hashicorp/boundary/internal/credential/vault/store"
+	"github.com/hashicorp/boundary/internal/db"
+	dbassert "github.com/hashicorp/boundary/internal/db/assert"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/iam"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/oplog"
+	"github.com/hashicorp/boundary/internal/scheduler"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRepository_CreateSSHCertificateCredentialLibrary(t *testing.T) {
+	t.Parallel()
+	conn, _ := db.TestSetup(t, "postgres")
+	rw := db.New(conn)
+	wrapper := db.TestWrapper(t)
+
+	_, prj := iam.TestScopes(t, iam.TestRepo(t, conn, wrapper))
+	cs := TestCredentialStores(t, conn, wrapper, prj.GetPublicId(), 1)[0]
+
+	tests := []struct {
+		name    string
+		in      *SSHCertificateCredentialLibrary
+		opts    []Option
+		want    *SSHCertificateCredentialLibrary
+		wantErr errors.Code
+	}{
+		{
+			name:    "nil-SSHCertificateCredentialLibrary",
+			wantErr: errors.InvalidParameter,
+		},
+		{
+			name:    "nil-embedded-SSHCertificateCredentialLibrary",
+			in:      &SSHCertificateCredentialLibrary{},
+			wantErr: errors.InvalidParameter,
+		},
+		{
+			name: "invalid-no-store-id",
+			in: &SSHCertificateCredentialLibrary{
+				SSHCertificateCredentialLibrary: &store.SSHCertificateCredentialLibrary{},
+			},
+			wantErr: errors.InvalidParameter,
+		},
+		{
+			name: "invalid-public-id-set",
+			in: &SSHCertificateCredentialLibrary{
+				SSHCertificateCredentialLibrary: &store.SSHCertificateCredentialLibrary{
+					StoreId:  cs.GetPublicId(),
+					PublicId: "abcd_OOOOOOOOOO",
+				},
+			},
+			wantErr: errors.InvalidParameter,
+		},
+		{
+			name: "invalid-no-username",
+			in: &SSHCertificateCredentialLibrary{
+				SSHCertificateCredentialLibrary: &store.SSHCertificateCredentialLibrary{
+					StoreId:   cs.GetPublicId(),
+					VaultPath: "/some/path",
+				},
+			},
+			wantErr: errors.InvalidParameter,
+		},
+		{
+			name: "invalid-no-vault-path",
+			in: &SSHCertificateCredentialLibrary{
+				SSHCertificateCredentialLibrary: &store.SSHCertificateCredentialLibrary{
+					StoreId:   cs.GetPublicId(),
+					VaultPath: "/some/path",
```

Comment:
This looks like it is setting a vault path, despite the test name being "invalid-no-vault-path"